tests/lib/ocpp/v201/test_profile.cpp

```diff
@@ -0,0 +1,1155 @@
+    DateTime now = dt("12:00");
+    DateTime end = dt("12:10");
+    std::vector<period_entry_t> combined_schedules{
+        {dt("12:05"), dt("13:00"), 32.0, 1, std::nullopt, 21, ChargingRateUnitEnum::A, std::nullopt},
+        {dt("11:30"), dt("12:30"), 24.0, 3, std::nullopt, 31, ChargingRateUnitEnum::A, std::nullopt}};
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{.startPeriod = 0, .limit = 24.0, .numberPhases = 3}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = now,
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, now, end, std::nullopt);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_NoGapA) {
+    DateTime now = dt("12:00");
+    DateTime end = dt("12:10");
+    std::vector<period_entry_t> combined_schedules{
+        {dt("11:50"), dt("12:05"), 32.0, 1, std::nullopt, 21, ChargingRateUnitEnum::A, std::nullopt},
+        {dt("12:05"), dt("12:30"), 24.0, 3, std::nullopt, 31, ChargingRateUnitEnum::A, std::nullopt}};
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{.startPeriod = 0, .limit = 32.0, .numberPhases = 1},
+                                   ChargingSchedulePeriod{.startPeriod = 300, .limit = 24.0, .numberPhases = 3}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = now,
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, now, end, std::nullopt);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_NoGapB) {
+    DateTime now = dt("12:00");
+    DateTime end = dt("12:10");
+    std::vector<period_entry_t> combined_schedules{
+        {dt("12:05"), dt("12:30"), 32.0, 1, std::nullopt, 21, ChargingRateUnitEnum::A, std::nullopt},
+        {dt("11:50"), dt("12:05"), 24.0, 3, std::nullopt, 31, ChargingRateUnitEnum::A, std::nullopt}};
+    CompositeSchedule expected = CompositeSchedule{
+        .chargingSchedulePeriod = {ChargingSchedulePeriod{.startPeriod = 0, .limit = 24.0, .numberPhases = 3},
+                                   ChargingSchedulePeriod{.startPeriod = 300, .limit = 32.0, .numberPhases = 1}},
+        .evseId = EVSEID_NOT_SET,
+        .duration = std::chrono::duration_cast<std::chrono::seconds>(minutes(10)).count(),
+        .scheduleStart = now,
+        .chargingRateUnit = ChargingRateUnitEnum::A,
+    };
+
+    CompositeSchedule actual = calculate_composite_schedule(combined_schedules, now, end, std::nullopt);
+
+    ASSERT_EQ(expected, actual);
+}
+
+TEST(OCPPTypesTest, CalculateChargingSchedule_OverlapA) {
```

Comment:
what is overlap A and how is it different from overlap B? None of the variables here are called A or B!